md-setup/src/main/java/com/wipro/ats/bdre/md/setup/SetupDB.java

```diff
@@ -402,15 +440,19 @@ private void populateProcessTemplate(String dataFile) throws Exception {
                     session.save(processTemplate);
                 }
             }
-            //session.flush();;
-        } catch (Exception e) {
-            LOGGER.error("In File: " + dataFile + "; Bad Line: " + line);
+        } catch (MetadataException e) {
+            LOGGER.error(inFile + dataFile + badLine + line);
             LOGGER.error(e.getMessage());
-            throw new Exception(e);
+            throw new MetadataException(e);
+        }
+        catch (IOException e) {
+            LOGGER.error(inFile + dataFile + badLine + line);
+            LOGGER.error(e.getMessage());
+            throw new IOException(e);
         }
     }
 
-    private void populateProcess(String dataFile) throws Exception {
+    private void populateProcess(String dataFile) throws MetadataException, IOException {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) The Cyclomatic Complexity of this method "populateProcess" is 13 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)